src/db.c

```diff
@@ -46,6 +48,76 @@ void updateLFU(robj *val) {
     val->lru = (LFUGetTimeInMinutes()<<8) | counter;
 }
 
+/* 
+ * Update histogram of keys-sizes
+ * 
+ * It is used to track the distribution of key sizes in the dataset. It is updated 
+ * every time key's length is modified. Available to user via INFO command. 
+ * 
+ * The histogram is a base-2 logarithmic histogram, with 64 bins. The i'th bin 
+ * represents the number of keys with a size in the range 2^i and 2^(i+1) 
+ * exclusive. oldLen/newLen must be smaller than 2^63. Each data type has its
+ * own histogram and it is per database (In addition, there is histogram per 
+ * slot for future use).
+ * 
+ * Examples to LEN values and corresponding bins in histogram: 
+ *               [1,2)->0 [2,4)->1 [4,8)->2 [8,16)->3
+ */
+void updateKeysizesHist(redisDb *db, int didx, uint32_t type, uint64_t oldLen, uint64_t newLen) {
+    serverAssert(unlikely(type < OBJ_TYPE_BASIC_MAX));
+
```

Comment:
should we exit when the old bin is equal to new bin?
in most case, they shouldn't change bin frequently.

Reply:
Most of the calls to `updateKeysizesHist` are being selectively applied following of a modification. I wouldn't be bothered to opt this scenario.

Reply:
If a dict has a length of 8, it means that the following 7 inserts are NOP and are increasing and decreasing at the same bin.
am i missing something?

Reply:
You are not missing anything. Except that once we evaluate the bing, most of the "computation" is beyond us. So we can either add another condition to verify if they are equal or just let the flow make ++ and -- blindly.

Reply:
i thought you just wanted to save some memory.

Reply:
well, since there isn't any regression, we can leave it as it is.